src/main/java/org/generationcp/middleware/dao/AttributeDAO.java

```diff
@@ -119,70 +87,69 @@
 		return returnList;
 	}
 
-	@SuppressWarnings("unchecked")
-	public List<UserDefinedField> getAttributeTypes() {
-		final Criteria criteria = this.getSession().createCriteria(UserDefinedField.class).add(Restrictions.eq("ftable", "ATRIBUTS"));
-		return criteria.list();
-	}
-
-	public Attribute getAttribute(final Integer gid, final String attributeName) {
+	public Attribute getAttributeByGidAndVariableId(final Integer gid, final Integer variableId) {
```

Comment:
This can reuse `getAttributeValuesByTypeAndGIDList` with singletonList of gid

Reply:
done